tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/localedistance/DistanceTable.java

```diff
@@ -0,0 +1,556 @@
+                // Default distances are the distance between any two *different* unknown
+                // subtags (so if the subtags are the same, the distance is zero).
+                int distance = key.getDesired().equals(key.getSupported()) ? 0 : wildcardMatch.distance;
+                node = new Node(distance);
+                node.copySubtablesFrom(wildcardMatch);
+                subtables.put(key.getDesired(), key.getSupported(), node);
+            }
+            return node;
+        }
+
+        /** Copies all subtable mappings from the given node into this one. */
+        private void copySubtablesFrom(Node src) {
+            checkState(subtables.isEmpty());
+            src.subtables.cellSet().forEach(
+                    c -> subtables.put(c.getRowKey(), c.getColumnKey(), new Node(c.getValue().distance)));
+        }
+
+        /**
+         * Writes all the mappings in the distance table sequentially to given Trie in sorted
+         * table order.
+         *
+         * <p>Mappings are written in a top-down recursive visitation with sub-tables inheriting
+         * the current prefix from parent tables via the given Trie span. At each level any
+         * mapped distances are written before recursing into the sub-tables.
+         */
+        private void writeTo(Span trie) {
+            if (distance >= 0 && (distance & DISTANCE_SKIP_SCRIPT) != 0) {
+                // If a node has no distance set, or has been explicitly marked as "skippable",
+                // then write the "default" subtable using the current Trie prefix (effectively
+                // having an "empty" prefix for this case).
+                getAnyNode().writeTo(trie);
+            } else {
+                // In the normal case, just write the mappings explicitly.
+                subtables.rowMap().forEach(
+                        (desired, supportedNodes) -> writeSupported(trie, desired, supportedNodes));
+            }
+        }
+
+        private void writeSupported(Span trie, String desired, Map<String, Node> supportedNodes) {
+            // Collapse any (desired=ANY, supported=ANY) mappings into a single '*' in the trie.
+            if (desired.equals(ANY)) {
+                // If desired is ANY, the only supported subtag must also be ANY.
+                Node node = supportedNodes.get(ANY);
+                checkState(node != null && supportedNodes.size() == 1,
+                        "invalid supported subtags for desired='ANY': %s", supportedNodes);
+                // Remember that ANY != "*", even though it corresponds to "*" in the original
+                // language match rules. Putting "*" in a Trie means something different (but
+                // similar enough to be a bit confusing).
+                trie.with("*", node::writeDistancePlusSubtables);
+            } else {
+                // In the general case, just write the <desired,supported> distance mapping.
+                trie.with(desired, withDesired ->
```

Comment:
`withDesiredSpan`

Reply:
Not sure "desired" helps here, but "withSpan" is good.

Reply:
I meant to ask for renaming the lambda function parameter, not the Trie member function.